src/wallet/wallet.h

```diff
@@ -487,6 +487,9 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
 
     // Used to prevent concurrent calls to walletpassphrase RPC.
     Mutex m_unlock_mutex;
+    // Used to prevent deleting the passphrase from memory when it is still in use.
+    RecursiveMutex m_relock_mutex;
```

Comment:
Using `Mutex` is preferable to `RecursiveMutex`.

Reply:
`Mutex` isn't used here because sometimes `m_relock_mutex` has to be locked multiple time within a scope to prevent deadlocks. See: https://github.com/bitcoin/bitcoin/pull/26347#discussion_r1006353769

> The new mutex is now grabbed in CWallet::Lock and also in locations where cs_wallet would have been grabbed first in order to prevent a deadlock.